tests/test_comparison_template_lib.py

```diff
@@ -162,3 +163,118 @@ def test_date_comparison_error_logger(ctl):
     # Metric len == 0
     with pytest.raises(ValueError):
         ctl.date_comparison("date", datediff_thresholds=[1], datediff_metrics=[])
+
+
+## name_comparison
+
+
+@pytest.mark.parametrize(
+    ("ctl"),
+    [
+        pytest.param(ctld, id="DuckDB Name Comparison Simple Run Test"),
+        pytest.param(ctls, id="Spark Name Comparison Simple Run Test"),
+    ],
+)
+def test_name_comparison_run(ctl):
+    ctl.name_comparison("date")
+
+
+@pytest.mark.parametrize(
+    ("ctl", "Linker"),
+    [
+        pytest.param(ctld, DuckDBLinker, id="DuckDB Date Comparison Integration Tests"),
+        pytest.param(ctls, SparkLinker, id="Spark Date Comparison Integration Tests"),
+    ],
+)
+def test_name_comparison_levels(spark, ctl, Linker):
+
+    df = pd.DataFrame(
+        [
+            {
+                "unique_id": 1,
+                "first_name": "Robert",
+                "first_name_metaphone": "RBRT",
+                "dob": "1996-03-25",
+            },
+            {
+                "unique_id": 2,
+                "first_name": "Rob",
+                "first_name_metaphone": "RB",
+                "dob": "1996-03-25",
+            },
+            {
+                "unique_id": 3,
+                "first_name": "Robbie",
+                "first_name_metaphone": "RB",
+                "dob": "1999-12-28",
+            },
+            {
+                "unique_id": 4,
+                "first_name": "Bobert",
+                "first_name_metaphone": "BB",
+                "dob": "2000-01-01",
+            },
+            {
+                "unique_id": 5,
+                "first_name": "Bobby",
+                "first_name_metaphone": "BB",
+                "dob": "2000-10-20",
+            },
+            {
+                "unique_id": 6,
+                "first_name": "Robert",
+                "first_name_metaphone": "RBRT",
+                "dob": "1996-03-25",
+            },
+        ]
+    )
+
+    settings = {
+        "link_type": "dedupe_only",
+        "comparisons": [
+            ctl.name_comparison(
+                "first_name",
+                phonetic_col_name="first_name_metaphone",
+            )
+        ],
+    }
+
+    if Linker == SparkLinker:
+        df = spark.createDataFrame(df)
+        df.persist()
+    linker = Linker(df, settings)
+    linker_output = linker.predict().as_pandas_dataframe()
+
+    # # Dict key: {gamma_level value: size}
+    size_gamma_lookup = {0: 8, 1: 4, 2: 0, 3: 2, 4: 1}
```

Comment:
maybe a little comment here outlining what each gamma level corresponds to, especially as otherwise have to check the default values and work out which level is which. Would also make things easier to update if we for some reason end up changing the defaults

Reply:
Yep that is sensible, have added some